Sources/MessagingPush/MessagingPush.swift

```diff
@@ -20,124 +20,44 @@ public protocol MessagingPushInstance: AutoMockable {
 public class MessagingPush: MessagingPushInstance {
     @Atomic public private(set) static var shared = MessagingPush(customerIO: CustomerIO.shared)
 
-    public let customerIO: CustomerIO!
-    private let httpClient: HttpClient
-    private let jsonAdapter: JsonAdapter
-    private let eventBus: EventBus
-
-    private var identifyCustomerEventBusCallback: NSObjectProtocol?
-
-    @Atomic public var deviceToken: String?
-
-    /// testing init
-    internal init(customerIO: CustomerIO?, httpClient: HttpClient, jsonAdapter: JsonAdapter, eventBus: EventBus) {
-        self.customerIO = customerIO ?? CustomerIO(siteId: "fake", apiKey: "fake", region: Region.EU)
-        self.httpClient = httpClient
-        self.jsonAdapter = jsonAdapter
-        self.eventBus = eventBus
-    }
+    public let customerIO: CustomerIOInstance!
+    internal var implementation: MessagingPushImplementation?
 
     /**
      Create a new instance of the `MessagingPush` class.
 
      - Parameters:
        - customerIO: Instance of `CustomerIO` class.
      */
-    public init(customerIO: CustomerIO) {
+    public init(customerIO: CustomerIOInstance) {
         self.customerIO = customerIO
-        self.httpClient = CIOHttpClient(credentials: customerIO.credentials!, config: customerIO.sdkConfig)
-        self.jsonAdapter = DITracking.shared.jsonAdapter
-        self.eventBus = DITracking.shared.eventBus
-
-        self.identifyCustomerEventBusCallback = eventBus.register(event: .identifiedCustomer) {
-//            if let deviceToken = self.deviceToken {
-            // register device token with customer.
-//            }
+        if let siteId = customerIO.siteId {
```

Comment:
mostly consistent with how our other API clients work - they only error once they're actually used - but would it be beneficial to developers to log or emit something at this point if the siteId doesn't exist?